File: Daily Test/Java/Maximum Events - Years And Months 24.6.2025.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        sc.nextLine();

        Map<String, Integer> yearMap = new HashMap<>();
        Map<String, Integer> monthMap = new HashMap<>();
        String[] months = {"Jan","Feb","Mar","Apr","May","Jun","Jul","Aug","Sep","Oct","Nov","Dec"};

        for (int i = 0; i < n; i++) {
            String[] parts = sc.nextLine().split("-");
            String mo = parts[1];
            String yr = parts[2];
            yearMap.put(yr, yearMap.getOrDefault(yr, 0) + 1);
            monthMap.put(mo, monthMap.getOrDefault(mo, 0) + 1);
        }

        int maxY = Collections.max(yearMap.values());
        int maxM = Collections.max(monthMap.values());

        List<Integer> sortedYears = new ArrayList<>();
        for (String y : yearMap.keySet()) {
            if (yearMap.get(y) == maxY) {
                sortedYears.add(Integer.parseInt(y));
            }
        }
        Collections.sort(sortedYears);
        for (int y : sortedYears) System.out.print(y + " ");
        System.out.println();

        for (String m : months) {
            if (monthMap.containsKey(m) && monthMap.get(m) == maxM) {
                System.out.print(m + " ");
            }
        }
    }
}

File: Daily Test/Python/Maximum Events - Years And Months 24.6.2025.py
n = int(input())
y = {}
m = {}
mon_order = ['Jan','Feb','Mar','Apr','May','Jun','Jul','Aug','Sep','Oct','Nov','Dec']

for _ in range(n):
    d = input().strip()
    _, mo, yr = d.split('-')
    y[yr] = y.get(yr, 0) + 1
    m[mo] = m.get(mo, 0) + 1

my = max(y.values())
mm = max(m.values())

ry = sorted([k for k in y if y[k] == my], key=int)
rm = [mon for mon in mon_order if mon in m and m[mon] == mm]

print(' '.join(ry))
print(' '.join(rm))